efel/pyfeatures/pyfeatures.py

```diff
@@ -28,39 +28,62 @@
 
 import numpy
 import efel.cppcore
+from scipy.ndimage.filters import gaussian_filter1d
+from numpy.fft import *
 
 
 all_pyfeatures = [
     'voltage',
     'time',
-    'current',
     'ISIs',
     'initburst_sahp',
     'initburst_sahp_vb',
     'initburst_sahp_ssse',
-    'depol_block']
+    'depol_block',
+    'current',
+    'impedance']
 
 
 def voltage():
     """Get voltage trace"""
     return _get_cpp_feature("voltage")
 
 
+def current():
+    """Get voltage trace"""
+    return _get_cpp_feature("current")
+
+
 def time():
     """Get time trace"""
     return _get_cpp_feature("time")
 
 
-def current():
-    """Get current trace"""
-    return _get_cpp_feature("current")
+def impedance():
+    volta = _get_cpp_feature("voltage")
+    volt_hold = numpy.median(volta[0:10])
+    volt = volta - volt_hold
+    curr = _get_cpp_feature("current")
+    cur_hold = numpy.median(curr[0:10])
+    cur = curr - cur_hold
+    if max(volta) < 0:  # if there is no spikes in ZAP
+        fft_volt = numpy.fft.fft(volt * 1e3)
+        fft_cur = numpy.fft.fft(cur * 1e12)
+        freq = numpy.fft.fftfreq(len(volt), d=0.0001)
+        Z = ((fft_volt) / (fft_cur))
```

Comment:
catch division by 0

Reply:
How should it behave upon a division by zero?

Reply:
Right now, `numpy` is raising a `RuntimeWarning`
` RuntimeWarning: divide by zero encountered in true_divide`

Reply:
Ok, a warning is ok, but it should never crash